src/azeventgrid.ngrok.webapi/Controllers/EventGridController.cs

```diff
@@ -29,19 +28,16 @@ public async Task<IActionResult> Post()
                 using (var reader = new StreamReader(Request.Body, Encoding.UTF8))
                 {
                     var jsonContent = await reader.ReadToEndAsync();
-                    var eventGridEvents = JsonConvert.DeserializeObject<List<EventGridEvent>>(jsonContent);
+                    var eventGridEvents = JsonSerializer.Deserialize<List<EventGridEvent>>(jsonContent);
 
                     foreach (var eventGridEvent in eventGridEvents)
                     {
                         // EventGrid validation message
-                        if (eventGridEvent.EventType == EventTypes.EventGridSubscriptionValidationEvent)
+                        if (eventGridEvent.EventType == "Microsoft.EventGrid.SubscriptionValidationEvent")
```

Comment:
It's a shame that [this](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventTypes.cs) was marked `internal`.    

@JoshLove-msft:  Any thoughts here?

Reply:
we can do a type check as shown [here](https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/eventgrid/Azure.Messaging.EventGrid#receiving-and-deserializing-events).
```cs
foreach (EventGridEvent egEvent in egEvents)
{
    // If the event is a system event, GetData() should return the correct system event type
   if (egEvent.GetData() is SubscriptionValidationEventData subscriptionData) 
  {
     return Ok(subscriptionData.ValidationCode);
  }
}
```

Reply:
Thanks!   I figured that I was overlooking something.